### PR TITLE
Refactor `merge_production_dotenvs_in_dotenv.py`

### PR DESCRIPTION
**Pull request by browniebroke**
## Description

Following up from #4086 and #4102, I wanted to rename a few more variables and then realised that this file could be modernised since it uses Pathlib. 

The tests was pretty much reimplementing the same logic as the function, and I think it would benefit from being "dumber" and simply write a bunch of input and compare it with the extected output.

Checklist:

- [x] I've made sure that tests are updated accordingly (especially if adding or updating a template option)
- [x] I've updated the documentation or confirm that my change doesn't require any updates

I think we could look at replacing this script by more standard tools, but that's probably one for another day:

- On Linux and macOS, it's pretty simple to do `cat file1 file2 ... > .env`  
- On Windows, I found this, but I have no idea if that's a viable option: https://www.shellhacks.com/windows-cat-equivalent-cmd-powershell/

## Rationale

Modernize script and make it more readable.

### CODE REVIEW

**Comment by luzfcb** [comment]
@browniebroke can we move the tests to another file? I guess it shouldn't be mandatory to have pytest installed to run this specific file

**Comment by browniebroke** [comment]
> @browniebroke can we move the tests to another file? I guess it shouldn't be mandatory to have pytest installed to run this specific file

Yes I wanted to do that but I think there might be some imports issue I'll need to address.

**Comment by browniebroke** [comment]
Another benefit of moving to a separate file: tests are [actually running](https://github.com/cookiecutter/cookiecutter-django/actions/runs/4019996894/jobs/6907458459#step:5:2436) now ([compared to master](https://github.com/cookiecutter/cookiecutter-django/actions/runs/4018856764/jobs/6904986697#step:5:2422))

**Comment by browniebroke** [comment]
@luzfcb WDYT? I think it's ready to go now. Let me know if you have any other suggestions, otherwise I'll merge it a bit later

**Review by luzfcb — APPROVED** [review]
LGTM 👍🏽

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -293,6 +293,7 @@ def set_flags_in_settings_files():
 def remove_envs_and_associated_files():
     shutil.rmtree(".envs")
     os.remove("merge_production_dotenvs_in_dotenv.py")
+    shutil.rmtree("tests")
```
Comment (luzfcb) [inline hooks/post_gen_project.py:296]: Why not remove /tests/test_merge_production_dotenvs_in_dotenv.py only?

**Review by luzfcb — COMMENTED** [review]
.

File: hooks/post_gen_project.py (hunk 1 above)
Comment (browniebroke) [reply hooks/post_gen_project.py:296]: That should be the only file in there. It's a new tests directory at the top level.

**Pull request merged**